### PR TITLE
Merge release 0.0.40

### PR DESCRIPTION
**Pull request by AlexGilleran**
### What this PR does
- Merges release 0.0.40
- Fixes national map links pointing at `/` instead of the external magda api url.

### CODE REVIEW

File: deploy/helm/magda-dev.yml [hunk 1]
```diff
@@ -131,7 +131,6 @@ correspondence-api:
     requests:
       cpu: 100m
       memory: 0
-  externalUrl: "https://magda-dev.terria.io"
```
Comment (t83714) [inline deploy/helm/magda-dev.yml]: Will this cause any trouble for dev site deployment (master branch)?
It seems `externalUrl` now is from `externalUrl=http://$CI_COMMIT_REF_SLUG.magda-dev.terria.io`
Does that mean dev site won't get value "https://magda-dev.terria.io" for `externalUrl`?
`Open in National Button` seems only work with `https` URL for cross-domain request

File: magda-web-server/src/index.ts [hunk 2]
```diff
@@ -146,6 +146,7 @@ app.get("/server-config.js", function(req, res) {
         disableAuthenticationFeatures: argv.disableAuthenticationFeatures,
         baseUrl: addTrailingSlash(argv.baseUrl),
         apiBaseUrl: apiBaseUrl,
+        baseExternalUrl: argv.baseExternalUrl,
```
Comment (t83714) [inline magda-web-server/src/index.ts]: Do we need `addTrailingSlash` it like `baseExternalUrl: addTrailingSlash(argv.baseExternalUrl)`?

File: deploy/helm/magda-dev.yml (hunk 1 above)
Comment (AlexGilleran) [reply deploy/helm/magda-dev.yml]: Dev site should be ok - externalUrl is still present on line 4, I've just made correspondence api use that one.

Would probably be wise to override it for the ui-only previews though 🤔

Comment (t83714) [reply deploy/helm/magda-dev.yml]: I think it's good enough then~ 👍

File: magda-web-server/src/index.ts (hunk 2 above)
Comment (AlexGilleran) [reply magda-web-server/src/index.ts]: Well, it works like it is, but I suppose it'll work with a trailing slash too - I'll add one and see.

File: deploy/helm/magda-dev.yml (hunk 1 above)
Comment (t83714) [reply deploy/helm/magda-dev.yml]: Probably don't need to override it as I guess one day gitlab preview will support `https`

**Pull request merged**